tool/nixvirt.py

```diff
@@ -128,6 +128,9 @@ def XMLDesc(self):
         return self.oc.XMLDesc(self.lvobj)
 
     def undefine(self):
+        isPersistent = self.lvobj.isPersistent()
```

Comment:
Have to get the persistence state before stopping the machine as after the machine is stopped it will throw the same error trying to retrieve it.